### DISCUSSION TITLE
How to add contour?

### DISCUSSION THREAD

**Discussion by 1923trk** [libvips/libvips#4011] [General]
Hello, Thank you for the library.
I have png images with alpha channels. I want to add contours to these images and get a result like the examples I will share below, but I couldn't find how to do it. Additionally I should be able to control the background color.
![54](https://github.com/kleisauke/net-vips/assets/172930587/db260337-be94-4e5d-90d5-5bed5cebdd19)
![24061513YKHV-01cqkdy8yiiputopi8le3xut](https://github.com/kleisauke/net-vips/assets/172930587/83f391d7-bfd7-4246-9453-89e29c4bce5d)
![169](https://github.com/kleisauke/net-vips/assets/172930587/77d32b28-24c6-47c9-9687-e9b82f3d3319)
![24061513YKGS-t7qy707avbs1bx3p82webwwq](https://github.com/kleisauke/net-vips/assets/172930587/816d2eba-2342-4835-898c-e73e8601150a)

**Comment by kipcole9** [comment]
I think this should move to a discussion rather than an issue.  

I'm no expert but I believe the approach you can take is this:

1. Take your alpha band and dilate it. Dilation will add edge pixels to represent the contour.
2. Take a copy of this dilated alpha band and colour it any way you want
3. Compose the base image over the coloured contour image
4. Add the dilated alpha band to re-establish the transparency.

I can see in your examples you have both a white border and then a feathered dark edge. You can use variations of the above to use more composed layers to get that effect. For example, to get the feathered edge, you can duplicate the dilated alpha band again and use a gaussian blur on it.

I'm sure John and Kleis can be more precise but they are in difference time zones - please take the above as a possible path from someone who is also learning how to do this kind of thing.

**Comment by jcupitt** [comment]
Exactly right Kip. I had a quick go in nip2 (the libvips GUI):

![image](https://github.com/libvips/libvips/assets/580843/ca7c8ed0-4ea0-44cc-9879-fc2435d844ff)

To make:

![sticker2](https://github.com/libvips/libvips/assets/580843/51cbc661-f1ee-4de6-84b4-df11df5ab929)

Workspace file here:

http://www.rollthepotato.net/~john/sticker.ws

Drop any image into A1 to make it into a sticker.
